pandas/tests/extension/base/methods.py

```diff
@@ -105,6 +105,31 @@ def test_factorize_equivalence(self, data_for_grouping, na_sentinel):
         tm.assert_numpy_array_equal(l1, l2)
         self.assert_extension_array_equal(u1, u2)
 
+    def test_fillna_copy_frame(self, data_missing):
+        arr = data_missing.take([1, 1])
+        df = pd.DataFrame({"A": arr})
+
+        filled_val = df.iloc[0, 0]
+        result = df.fillna(filled_val)
+
+        assert df.values.base is not result.values.base
```

Comment:
I don't think this is correct. I believe that for DataFrames with EAs, `df.values` will all return a new ndarray, so this would be impossible to fail.

Rather, I think the best we can do is `assert df.A.values is not result.A.values`.